modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java

```diff
@@ -429,6 +434,11 @@ private int _authenticate(
 				return FAILURE;
 			}
 
+			// Get user or create fromUnsafe LDAP
+
+			User user = _ldapUserImporter.importUser(
+				ldapServerId, companyId, safeLdapContext, attributes, password);
```

Comment:
Do you think we will never require to import a user from LDAP if the authentication fails?